src/lib-voyager.test.ui.tsx

```diff
@@ -224,5 +224,80 @@ describe('lib-voyager', () => {
 
       }, DEFAULT_TIMEOUT_LENGTH);
     });
+
+    it('getSpec returns Vega-Lite spec', done => {
+      setTimeout(() => {
+        try {
+          const voyagerInst = CreateVoyager(container, undefined, undefined);
+
+          const spec: Object = {
+            "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
+            "data": {
+              "values": [
+                {"date": "24-Apr-07", "close": "93.24"},
+                {"date": "25-Apr-07", "close": "95.35"},
+                {"date": "26-Apr-07", "close": "98.84"},
+                {"date": "27-Apr-07", "close": "99.92"}
+              ]
+            },
+            "mark": "bar",
+            "encoding": {
+              "x": {
+                "bin": true,
+                "field": "close",
+                "type": "quantitative"
+              },
+              "y": {
+                "aggregate": "count",
+                "field": "*",
+                "type": "quantitative"
+              }
+            }
+          };
+          voyagerInst.setSpec(spec);
+
+          setTimeout(() => {
+            try {
+              const retrievedSpec = voyagerInst.getSpec();
+              expect(retrievedSpec).toEqual({
+                data: {
+                  name: 'source'
```

Comment:
cc @kanitw: Every VL spec in the store has `data: {name: 'source'}`. Should other programs like David's receive the spec like this when they make a getSpec() API call?

Reply:
Great question, I think appending data (either url or inline) might make more sense?

Reply:
Good suggestion! I recall @davidanthoff mentioning the idea of having a boolean flag indicating whether or not to attach the data payload. I made this change in the next iteration.

Reply:
Yes, a flag is ideal there. I'm not sure, but the current version now would include the ``data`` element with the name of the source if the function is called with the flag to not include data? Or did I misunderstand that? I guess what I'm saying is that when the function is called with the flag to not include data, it would be great if the spec just didn't have any `data` element.

Reply:
> the current version now would include the data element with the name of the source if the function is called with the flag to not include data? 

Yes, that's the right behavior.  A spec without `data` is an invalid Vega-Lite spec and would make it fail to match output type.  This way you can directly bind data to the named data source. If you want to replace `data` with your own new data definition, as a user of the API, you can always replace `data` with what you want later.

Reply:
Ah, got it, that makes sense!